generators/csharp/base/src/asIs/JsonConfiguration.Template.cs

```diff
@@ -103,6 +102,46 @@ internal static JsonDocument SerializeToDocument<T>(T obj) =>
     internal static byte[] SerializeToUtf8Bytes<T>(T obj) =>
         JsonSerializer.SerializeToUtf8Bytes(obj, JsonOptions.JsonSerializerOptions);
 
+    internal static string SerializeWithAdditionalProperties<T>(T obj, object? additionalProperties = null)
+    {
+        if (additionalProperties == null)
+        {
+            return Serialize(obj);
+        }
+        var additionalPropertiesJsonNode = SerializeToNode(additionalProperties);
+        if (additionalPropertiesJsonNode is not JsonObject additionalPropertiesJsonObject)
+        {
+            throw new InvalidOperationException("The additional properties must serialize to a JSON object.");
+        }
+        var jsonNode = SerializeToNode(obj);
+        if (jsonNode is not JsonObject jsonObject)
+        {
+            throw new InvalidOperationException("The serialized object must be a JSON object to add properties.");
+        }
+        MergeJsonObjects(jsonObject, additionalPropertiesJsonObject);
+        return jsonObject.ToJsonString(JsonOptions.JsonSerializerOptions);
+    }
+
+    private static void MergeJsonObjects(JsonObject baseObject, JsonObject overrideObject)
+    {
+        foreach (var property in overrideObject)
+        {
+            if (!baseObject.TryGetPropertyValue(property.Key, out JsonNode? existingValue))
+            {
+                baseObject[property.Key] = property.Value != null ? JsonNode.Parse(property.Value.ToJsonString()) : null;
+                continue;
+            }
+            if (existingValue is JsonObject nestedBaseObject && property.Value is JsonObject nestedOverrideObject)
+            {
+                // If both values are objects, recursively merge them.
+                MergeJsonObjects(nestedBaseObject, nestedOverrideObject);
+                continue;
+            }
+            // Otherwise, the overrideObject takes precedence.
+            baseObject[property.Key] = property.Value != null ? JsonNode.Parse(property.Value.ToJsonString()) : null;
```

Comment:
Is `baseObject[property.Key] = property.Value != null ? JsonNode.Parse(property.Value.ToJsonString()) : null;` is duplicate, you can simplify this

Reply:
Discussed offline - going to leave as-is for now.